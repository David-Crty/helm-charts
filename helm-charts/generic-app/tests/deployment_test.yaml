# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json

suite: test deployment
release:
  name: test
chart:
  version: 0.1.0
templates:
  - deployment.yaml
tests:
  - it: should have init containers
    set:
      initContainers:
        - name: db-migrate
          command: [ 'bin/console', 'doctrine:migrations:migrate', '-n' ]
    asserts:
      - equal:
          path: spec.template.spec.initContainers[0]
          value:
            name: db-migrate-test-generic-app
            image: hello-world:latest
            imagePullPolicy: Always
            command:
              - bin/console
              - doctrine:migrations:migrate
              - '-n'
  - it: should have init containers with custom image
    set:
      initContainers:
        - name: db-migrate
          command: [ 'bin/console', 'doctrine:migrations:migrate', '-n' ]
          image: hello-world-innit
          tag: "123"
    asserts:
      - equal:
          path: spec.template.spec.initContainers[0]
          value:
            name: db-migrate-test-generic-app
            image: hello-world-innit:123
            imagePullPolicy: Always
            command:
              - bin/console
              - doctrine:migrations:migrate
              - '-n'
  - it: should have volume mounts for data
    set:
      volumes:
        data:
          size: 1Gi
          storageClass: ""
          mountPath: /data
    asserts:
      - matchSnapshot: { }
  - it: should have php production envs
    set:
      webdevopsPhpProductionEnvs: true
    asserts:
      - contains:
          path: spec.template.spec.containers[0].env
          content:
            name: php.opcache.validate_timestamps
            value: "0"
  - it: should have volume mounts for configmap
    set:
      volumes:
        cm:
          mountPath: /data
          configMap:
            name: my-configmap
    asserts:
      - matchSnapshot: { }
  - it: should have a worker pod
    set:
      env:
        TEST: test
      workers:
        consumer:
          command: [ 'bin/console', 'messenger:consume', 'async', '-vv' ]
          env:
            TEST_CONSUMER: test
    asserts:
      - matchSnapshot: { }
  - it: should have a security context
    set:
      securityContext:
        runAsUser: 1000
        capabilities:
          add:
            - NET_ADMIN
    asserts:
      - equal:
          path: spec.template.spec.containers[0].securityContext
          value:
            runAsUser: 1000
            capabilities:
              add:
                - NET_ADMIN
