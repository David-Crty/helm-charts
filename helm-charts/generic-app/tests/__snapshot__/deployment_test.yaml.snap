should have a worker pod:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        link.argocd.argoproj.io/external-link: https://
      labels:
        app.kubernetes.io/instance: test
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: generic-app
        helm.sh/chart: generic-app-0.1.0
      name: test-generic-app
    spec:
      replicas: 1
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app.kubernetes.io/component: web
          app.kubernetes.io/instance: test
          app.kubernetes.io/name: generic-app
      strategy:
        type: RollingUpdate
      template:
        metadata:
          annotations:
            checksum/env: 16892e71d2bf2d62ceab279ed42d021ff8d0bb0c3a6ad35677fdb74d7760092c
            checksum/envEncrypted: 52133ddfcd583280388a8bbfb67a417b6b660f75bc63ec1b01beec811d4498ef
          labels:
            app.kubernetes.io/component: web
            app.kubernetes.io/instance: test
            app.kubernetes.io/name: generic-app
        spec:
          containers:
            - envFrom:
                - configMapRef:
                    name: test-generic-app
              image: hello-world:latest
              imagePullPolicy: Always
              lifecycle:
                preStop:
                  exec:
                    command:
                      - sleep
                      - "10"
              name: test-generic-app
              ports:
                - containerPort: 80
                  name: http
              resources:
                limits:
                  ephemeral-storage: 2Gi
                  memory: 512Mi
                requests:
                  cpu: 50m
                  ephemeral-storage: 1Gi
                  memory: 128Mi
          serviceAccountName: default
  2: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: test
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: generic-app
        helm.sh/chart: generic-app-0.1.0
      name: test-generic-app-consumer
    spec:
      replicas: null
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app.kubernetes.io/component: worker
          app.kubernetes.io/instance: test
          app.kubernetes.io/name: generic-app
          app.kubernetes.io/name-worker: test-generic-app-consumer
      template:
        metadata:
          annotations:
            checksum/env: 16892e71d2bf2d62ceab279ed42d021ff8d0bb0c3a6ad35677fdb74d7760092c
            checksum/envEncrypted: 52133ddfcd583280388a8bbfb67a417b6b660f75bc63ec1b01beec811d4498ef
          labels:
            app.kubernetes.io/component: worker
            app.kubernetes.io/instance: test
            app.kubernetes.io/name: generic-app
            app.kubernetes.io/name-worker: test-generic-app-consumer
        spec:
          containers:
            - command:
                - bin/console
                - messenger:consume
                - async
                - -vv
              env:
                - name: TEST_CONSUMER
                  value: test
              envFrom:
                - configMapRef:
                    name: test-generic-app
              image: hello-world:latest
              imagePullPolicy: Always
              name: test-generic-app-consumer
              resources:
                limits:
                  ephemeral-storage: 2Gi
                  memory: 512Mi
                requests:
                  cpu: 50m
                  ephemeral-storage: 1Gi
                  memory: 128Mi
should have volume mounts for configmap:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        link.argocd.argoproj.io/external-link: https://
      labels:
        app.kubernetes.io/instance: test
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: generic-app
        helm.sh/chart: generic-app-0.1.0
      name: test-generic-app
    spec:
      replicas: 1
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app.kubernetes.io/component: web
          app.kubernetes.io/instance: test
          app.kubernetes.io/name: generic-app
      strategy:
        type: RollingUpdate
      template:
        metadata:
          annotations:
            checksum/env: 52133ddfcd583280388a8bbfb67a417b6b660f75bc63ec1b01beec811d4498ef
            checksum/envEncrypted: 52133ddfcd583280388a8bbfb67a417b6b660f75bc63ec1b01beec811d4498ef
          labels:
            app.kubernetes.io/component: web
            app.kubernetes.io/instance: test
            app.kubernetes.io/name: generic-app
        spec:
          containers:
            - image: hello-world:latest
              imagePullPolicy: Always
              lifecycle:
                preStop:
                  exec:
                    command:
                      - sleep
                      - "10"
              name: test-generic-app
              ports:
                - containerPort: 80
                  name: http
              resources:
                limits:
                  ephemeral-storage: 2Gi
                  memory: 512Mi
                requests:
                  cpu: 50m
                  ephemeral-storage: 1Gi
                  memory: 128Mi
              volumeMounts:
                - mountPath: /data
                  name: cm
          serviceAccountName: default
          volumes:
            - configMap:
                name: my-configmap
              name: cm
should have volume mounts for data:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        link.argocd.argoproj.io/external-link: https://
      labels:
        app.kubernetes.io/instance: test
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: generic-app
        helm.sh/chart: generic-app-0.1.0
      name: test-generic-app
    spec:
      replicas: 1
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app.kubernetes.io/component: web
          app.kubernetes.io/instance: test
          app.kubernetes.io/name: generic-app
      strategy:
        type: RollingUpdate
      template:
        metadata:
          annotations:
            checksum/env: 52133ddfcd583280388a8bbfb67a417b6b660f75bc63ec1b01beec811d4498ef
            checksum/envEncrypted: 52133ddfcd583280388a8bbfb67a417b6b660f75bc63ec1b01beec811d4498ef
          labels:
            app.kubernetes.io/component: web
            app.kubernetes.io/instance: test
            app.kubernetes.io/name: generic-app
        spec:
          containers:
            - image: hello-world:latest
              imagePullPolicy: Always
              lifecycle:
                preStop:
                  exec:
                    command:
                      - sleep
                      - "10"
              name: test-generic-app
              ports:
                - containerPort: 80
                  name: http
              resources:
                limits:
                  ephemeral-storage: 2Gi
                  memory: 512Mi
                requests:
                  cpu: 50m
                  ephemeral-storage: 1Gi
                  memory: 128Mi
              volumeMounts:
                - mountPath: /data
                  name: data
          serviceAccountName: default
          volumes:
            - name: data
              persistentVolumeClaim:
                claimName: test-generic-app-data
